src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -300,8 +319,92 @@ public void visitParameter(JdbcParameter jdbcParameter) {
     }
 
     @Override
+    @SuppressWarnings("OperatorPrecedence")
     public void visitSelectStatement(SelectStatement selectStatement) {
-        throw new FeatureNotSupportedException("TODO-HIBERNATE-22 https://jira.mongodb.org/browse/HIBERNATE-22");
+        logSqlAst(selectStatement);
+        if (!selectStatement.getQueryPart().isRoot()) {
+            throw new FeatureNotSupportedException("Subquery not supported");
+        }
+        if (selectStatement.getCteStatements() != null
+                        && !selectStatement.getCteStatements().isEmpty()
+                || selectStatement.getCteObjects() != null
+                        && !selectStatement.getCteObjects().isEmpty()) {
+            throw new FeatureNotSupportedException("CTE feature not supported");
+        }
```

Comment:
CTE (Common Table Expression; See https://hightouch.com/sql-dictionary/sql-common-table-expression-cte for a gentle introduction to CTE) is about creating some temporary table. Mongo aggregate command seems to support it, but for this PR let us throw exception for now.